internal/service/advancedcluster/resource_advanced_cluster_test.go

```diff
@@ -1181,7 +1186,17 @@ func checkReplicaSetMultiCloud(name string, regionConfigs int) resource.TestChec
 	)
 }
 
-func configShardedOldSchemaMultiCloud(orgID, projectName, name string, numShards int, analyticsSize string) string {
+func configShardedOldSchemaMultiCloud(orgID, projectName, name string, numShards int, analyticsSize string, configServerManagementMode *string) string {
+	var rootConfig string
+	if configServerManagementMode != nil {
+		// valid values: FIXED_TO_DEDICATED or ATLAS_MANAGED (default)
+		// only valid for Major version 8 and later
+		// cluster must be SHARDED
+		rootConfig = fmt.Sprintf(`
+		  mongo_db_major_version = "8"
+		  config_server_management_mode = %[1]q
```

Comment:
should we clarify in docs that this attribute requires mongodb version > 8 and cluster must be sharded?

Reply:
You can find the doc updates here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2677 it mentions shared.
About version > 8, that is only for embedded, so the field is still valid on earlier versions but it will always be direct.